ui/src/app/workflows/components/resubmit-workflow-panel.tsx

```diff
@@ -12,103 +12,93 @@ interface Props {
+    const [memoized, setMemoized] = useState(false);
+    const [error, setError] = useState<Error>();
+    const [isSubmitting, setIsSubmitting] = useState(false);
 
-export class ResubmitWorkflowPanel extends React.Component<Props, State> {
-    constructor(props: any) {
-        super(props);
-        const state: State = {
-            workflowParameters: JSON.parse(JSON.stringify(this.props.workflow.spec.arguments.parameters || [])),
-            memoized: false,
-            isSubmitting: false,
-            overrideParameters: false
+    const submit = () => {
+        setIsSubmitting(true);
+        const parameters: ResubmitOpts['parameters'] = overrideParameters
+            ? [...workflowParameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p))]
+            : [];
+        const opts: ResubmitOpts = {
+            parameters,
+            memoized
         };
-        this.state = state;
-    }
 
-    public render() {
-        return (
-            <>
-                <h4>Resubmit Workflow</h4>
-                <h5>
-                    {this.props.workflow.metadata.namespace}/{this.props.workflow.metadata.name}
-                </h5>
-                {this.state.error && <ErrorNotice error={this.state.error} />}
-                <div className='white-box'>
-                    <div key='override-parameters' style={{marginBottom: 25}}>
-                        <label>Override Parameters</label>
-                        <div className='columns small-9'>
-                            <Checkbox checked={this.state.overrideParameters} onChange={overrideParameters => this.setState({overrideParameters})} />
-                        </div>
-                    </div>
+        const handleSuccess = (submitted: Workflow) => {
+            document.location.href = uiUrl(`workflows/${submitted.metadata.namespace}/${submitted.metadata.name}`);
+        };
 
-                    {this.state.overrideParameters && (
-                        <div key='parameters' style={{marginBottom: 25}}>
-                            <label>Parameters</label>
-                            {this.state.workflowParameters.length > 0 && (
-                                <ParametersInput parameters={this.state.workflowParameters} onChange={workflowParameters => this.setState({workflowParameters})} />
-                            )}
-                            {this.state.workflowParameters.length === 0 ? (
-                                <>
-                                    <br />
-                                    <label>No parameters</label>
-                                </>
-                            ) : (
-                                <></>
-                            )}
-                        </div>
-                    )}
+        const handleError = (err: Error) => {
+            setError(err);
+            setIsSubmitting(false);
+        };
```

Comment:
```suggestion
        function handleError(err: Error) {
            setError(err);
            setIsSubmitting(false);
        }
```

same as above

Reply:
or this gets removed entirely if `submit` is [rewritten](https://github.com/argoproj/argo-workflows/pull/11803#discussion_r1323481200) as it gets in-lined

Reply:
I reverted handleSuccess and handleError funcs to inline because there was no need to extract functions by using await/async and try/catch as you pointed out. thanks!